### PR TITLE
Np 47076 project contributors only added once

### PR DESCRIPTION
**Pull request by torarnet**
Changed datamodel for Project participants. PR has a lot of lines changed but it was not possible to keep it shorter than it is as it affects a lot of the code base for project module

### CODE REVIEW

File: cristin-project/src/test/java/no/unit/nva/cristin/projects/create/createProject.feature [hunk 1]
```diff
@@ -10,10 +10,96 @@ Feature: API tests for Cristin Project retrieve and search
...
-    * def requestWithIllegalValues =   { type: 'Project', title: 'Test Project', startDate: '2010-06-29T01:33:17.518Z', endDate: 'IllegalValue', coordinatingInstitution: {type: 'Organization',id: 'https://api.cristin-test.uio.no/v2/institutions/20202'},contributors: [{type: 'IllegalValue',identity: {type: 'Person',id: 'https://api.cristin-test.uio.no/v2/persons/515114'},affiliation: {type: 'Organization',id: 'https://api.cristin-test.uio.no/v2/institutions/20202'}}]}
-    * def requestWithId =   { id: 'https://api.dev.nva.aws.unit.no/cristin/project/2121331275', type: 'Project', title: 'Test Project', language: 'http://lexvo.org/id/iso639-3/nob', startDate: '2010-06-29T01:33:17.518Z', endDate: '2030-06-29T01:33:17.518Z', coordinatingInstitution: {type: 'Organization',id: 'https://api.cristin-test.uio.no/v2/institutions/20202'},contributors: [{type: 'ProjectManager',identity: {type: 'Person',id: 'https://api.cristin-test.uio.no/v2/persons/515114'},affiliation: {type: 'Organization',id: 'https://api.cristin-test.uio.no/v2/institutions/20202'}}]}
+    * def minimalCreateRequest =
```
Comment (torarnet) [inline cristin-project/src/test/java/no/unit/nva/cristin/projects/create/createProject.feature:13]: Better formatting for readability and changed to new datamodel

File: cristin-project/src/test/java/no/unit/nva/cristin/projects/create/createProject.feature [hunk 2]
```diff
@@ -45,16 +131,6 @@ Feature: API tests for Cristin Project retrieve and search
     When method POST
     Then status 401
 
-  Scenario: Create returns status Bad request when input has id
```
Comment (torarnet) [inline cristin-project/src/test/java/no/unit/nva/cristin/projects/create/createProject.feature:48]: Do not think this test is necessary anymore as we have another 400 test

File: cristin-project/src/test/java/no/unit/nva/cristin/projects/create/createSampleProject.feature [hunk 3]
```diff
@@ -30,36 +30,55 @@ Feature: API tests for creating sample Cristin Projects
...
-          'type': 'ProjectManager',
           'identity': {
             'type': 'Person',
             'id': 'https://api.dev.nva.aws.unit.no/cristin/person/1684651'
           },
-          'affiliation': {
-            'type': 'Organization',
-            'id': 'https://api.dev.nva.aws.unit.no/cristin/organization/20754.0.0.0'
-          }
+          'roles': [
```
Comment (torarnet) [inline cristin-project/src/test/java/no/unit/nva/cristin/projects/create/createSampleProject.feature:37]: Here is the main change. As one can see we now only have one identity per project, but several roles with affiliations per that identity. With the old version we could have duplicate identity causing us to have more contributor entries than people

File: cristin-project/src/test/java/no/unit/nva/cristin/projects/fetch/FetchCristinProjectHandlerTest.java [hunk 4]
```diff
@@ -273,19 +271,6 @@ void handlerThrowsNotAcceptableWhenAcceptHeaderUnsupported(String contentTypeReq
             containsString(String.format(ERROR_MESSAGE_UNSUPPORTED_CONTENT_TYPE, contentTypeRequested)));
     }
 
-    @Test
-    void handlerReturnsNvaProjectContainingFundingFromCristinWhenFundingHasValuesInCristin() throws Exception {
```
Comment (torarnet) [inline cristin-project/src/test/java/no/unit/nva/cristin/projects/fetch/FetchCristinProjectHandlerTest.java:277]: already is in json testdata asserted in another test. redundant and removed

File: cristin-project/src/test/java/no/unit/nva/cristin/projects/fetch/FetchCristinProjectHandlerTest.java [hunk 5]
```diff
@@ -129,25 +126,26 @@ void handlerReturnsBadGatewayWhenStatusCodeFromBackendSignalsError() throws Exce
...
     @Test
     void handlerReturnsNvaProjectFromTransformedCristinProjectWhenIdIsFound() throws Exception {
-        GatewayResponse<NvaProject> gatewayResponse = sendQueryWithId(DEFAULT_IDENTIFIER);
-        String expected = getBodyFromResource(API_RESPONSE_ONE_PROJECT_JSON);
-        assertEquals(OBJECT_MAPPER.readValue(expected, NvaProject.class),
-                gatewayResponse.getBodyObject(NvaProject.class));
+        var gatewayResponse = sendQueryWithId(DEFAULT_IDENTIFIER);
+        var expected = getBodyFromResource(API_RESPONSE_ONE_PROJECT_JSON);
+
+        JSONAssert.assertEquals(expected, gatewayResponse.getBody(), JSONCompareMode.NON_EXTENSIBLE);
```
Comment (torarnet) [inline cristin-project/src/test/java/no/unit/nva/cristin/projects/fetch/FetchCristinProjectHandlerTest.java:148]: compare strings directly with JSONAssert. Will not miss any (de)serialization issues

File: cristin-project/src/test/java/no/unit/nva/cristin/projects/query/QueryCristinProjectHandlerTest.java [hunk 6]
```diff
@@ -176,13 +178,11 @@ void setUp() {
         handler = new QueryCristinProjectHandler(clientProvider, environment);
     }
 
-    @ParameterizedTest
-    @MethodSource("queryResponseWithFundingFileReader")
-    void handlerReturnsExpectedBodyWhenRequestInputIsValid(String expected) throws IOException {
+    @Test
+    void handlerReturnsExpectedBodyWhenRequestInputIsValid() throws Exception {
```
Comment (torarnet) [inline cristin-project/src/test/java/no/unit/nva/cristin/projects/query/QueryCristinProjectHandlerTest.java:182]: Removing parameters as it was only one parameter

**Review by anetteOlli — APPROVED** [review]

File: cristin-project/src/test/java/no/unit/nva/cristin/projects/update/updateProject.feature [hunk 7]
```diff
@@ -23,11 +23,19 @@ Feature: API tests for Cristin Project Update
...
-          'type': 'ProjectManager',
           'identity': {
             'type': 'Person',
             'id': 'https://api.dev.nva.aws.unit.no/cristin/person/1684652'
-          }
+          },
+          'roles' : [
+            {
+              'type': 'ProjectManager',
+              'affiliation': {
```
Comment (torarnet) [inline cristin-project/src/test/java/no/unit/nva/cristin/projects/update/updateProject.feature:33]: add an affiliation for test coverage

**Review by brinxmat — APPROVED** [review]

**Pull request merged**